### PR TITLE
More tests are added for verifying TLS/SNI functionality.

### PR DESCRIPTION
**Pull request by andyuk1986**
Added more tests testing more failover scenarios. 
Testing multiple SNI identities.

### CODE REVIEW

File: spec/infinispan_ssl_spec.js [hunk 1]
```diff
@@ -97,6 +133,16 @@ describe('Infinispan TLS/SSL client', function() {
     }
   }
 
+  function sslSniTrusted1() {
```
Comment (galderz) [inline spec/infinispan_ssl_spec.js]: Can we change this name, along with the `trusted1` prefix into something more meaningful? What are you testing exactly?

File: spec/infinispan_ssl_spec.js [hunk 2]
```diff
@@ -160,4 +215,82 @@ describe('Infinispan TLS/SSL client', function() {
     }
   }
 
+  function sslStoreNoPath1() {
```
Comment (galderz) [inline spec/infinispan_ssl_spec.js]: Same here, please use more meaningful names.

File: spec/infinispan_ssl_spec.js [hunk 3]
```diff
@@ -31,6 +35,14 @@ describe('Infinispan TLS/SSL client', function() {
      testError('CERT_SIGNATURE_FAILURE', sslSniUntrusted())
   );
 
+  it('fails to operate if server name (SNI) is not provided, but certificate is trusted',
+      testError('CERT_SIGNATURE_FAILURE', sslSniDefaultWithTrustedCertificate())
+  );
+
+  it('fails to operate if server name (SNI) and trusted certificate are incorrect',
```
Comment (galderz) [inline spec/infinispan_ssl_spec.js]: Is this description representative of what the test does? The client does not seem to provide any trusted certificates here.

**Comment by galderz** [comment]
Minor comments about the naming of some things but otherwise looks good, thanks :)

File: spec/infinispan_ssl_spec.js (hunk 3 above)
Comment (andyuk1986) [inline spec/infinispan_ssl_spec.js]: Ooops, copy/paste issue. Thanks for highlighting. Fixed.

File: spec/infinispan_ssl_spec.js (hunk 1 above)
Comment (andyuk1986) [inline spec/infinispan_ssl_spec.js]: I have renamed all necessary functions. This test tests that if server has 2 trusted sni-s, then both can be used for the authentication.

**Comment by andyuk1986** [comment]
@galderz I have fixed your comments and squashed my changes to one commit. Please let me know if you have more comments.

**Comment by galderz** [comment]
Looks good, thanks for the changes 👍 

I might still rename the new keystore files further to better distinguish between them but I can do that with the changes coming up for domain (see #12). 

Thanks once again! :)

**Pull request merged**